src/rpc/request.cpp

```diff
@@ -134,8 +134,11 @@ static fs::path GetMainchainAuthCookieFile()
     if (gArgs.GetChainName() == "liquidv1") {
         cookie_file = ".cookie";
     }
-    std::string path = gArgs.GetArg("-mainchainrpccookiefile", cookie_file);
-    return AbsPathForConfigVal(fs::path(path));
+    std::string path = gArgs.GetArg("-mainchainrpccookiefile", "");
+    if (path == "") {
+        return fsbridge::AbsPathJoin(GetMainchainDefaultDataDir(), cookie_file);
+    }
+    return AbsPathForConfigVal(fs::path(cookie_file));
 }
```

Comment:
Deleted my previous comment, which was wrong. Let's try again.

This doesn't work right if `-mainchainrpccookiefile` is specified, which is presumably why CI is failing.

Instead, I would suggest something like this:

```
    if (gArgs.GetChainName() == "liquidv1") {
        cookie_file = ".cookie";
    }
    cookie_file = fsbridge::AbsPathJoin(GetMainchainDefaultDataDir(), cookie_file);
```

so that you are only changing how the default works, and leave the rest of the logic (handling user arguments with GetArg) alone.